internal/rcm/api.go

```diff
@@ -216,16 +205,7 @@ func (api *API) submit(writer http.ResponseWriter, request *http.Request, _ http
 	// Push the requested compose to the store
 	composeUUID := uuid.New()
 	// nil is used as an upload target, because LocalTarget is already used in the PushCompose function
-	err = api.store.PushComposeRequest(store.ComposeRequest{
-		Blueprint:       blueprint.Blueprint{},
-		ComposeID:       composeUUID,
-		Distro:          distro,
-		Arch:            composeRequest.Architectures[0],
-		Repositories:    repoConfigs,
-		Packages:        packages,
-		BuildPackages:   buildPackages,
-		RequestedImages: requestedImages,
-	})
+	err = api.store.PushCompose(distro, composeUUID, &blueprint.Blueprint{}, packages, buildPackages, arch, imageType, 0, nil)
```

Comment:
The `ComposeRequest` structure was supposed to represent what the user wanted. My intention here was to pass the requirements to the store and the store would do the depsolving etc. This way we are mixing the user input (blueprint) and our internal implementation (packages, build packages) which seems unnecessary to me. The `ComposeRequest` was also nice for decoupling the API logic from the logic in store.

Reply:
See below.